helm/nessie/templates/_helpers.tpl

```diff
@@ -273,44 +273,52 @@ Define environkent variables for catalog storage options.
 */}}
 {{- define "nessie.catalogStorageEnv" -}}
 {{ $global := .}}
-{{- include "nessie.secretToEnv" (list .Values.catalog.storage.s3.defaultOptions.accessKeySecret "awsAccessKeyId" "nessie.catalog.service.s3.default-options.access-key.name" . ) }}
-{{- include "nessie.secretToEnv" (list .Values.catalog.storage.s3.defaultOptions.accessKeySecret "awsSecretAccessKey" "nessie.catalog.service.s3.default-options.access-key.secret" . ) }}
+{{- include "nessie.secretToEnv" (list .Values.catalog.storage.s3.defaultOptions.accessKeySecret "awsAccessKeyId" "nessie-catalog-secrets.s3.default-options.access-key.name" "nessie.catalog.service.s3.default-options.access-key" "nessie-catalog-secrets.s3.default-options.access-key" . ) }}
+{{- include "nessie.secretToEnv" (list .Values.catalog.storage.s3.defaultOptions.accessKeySecret "awsSecretAccessKey" "nessie-catalog-secrets.s3.default-options.access-key.secret" "" "" . ) }}
```

Comment:
This seems to produce incorrect k8s yaml... for example:
```
            - name: "nessie-catalog-secrets.s3.default-options.access-key.name"
              valueFrom:
                secretKeyRef:
                  name: "k8sSecABC"
                  key: "key111"
            - name: "nessie.catalog.service.s3.default-options.access-key"
              value: "nessie-catalog-secrets.s3.default-options.access-key"- name: "nessie-catalog-secrets.s3.default-options.access-key.secret"
              valueFrom:
                secretKeyRef:
                  name: "k8sSecABC"
                  key: "access111"
```
The second one (secret) seems to be broken syntax-wise.

Also, `nessie-catalog-secrets.s3.default-options.access-key` is not referenced from the config map.

Test values (diff):
```
diff --git a/helm/nessie/values.yaml b/helm/nessie/values.yaml
index 3e066c45a..ace6e7f7f 100644
--- a/helm/nessie/values.yaml
+++ b/helm/nessie/values.yaml
@@ -138,14 +138,14 @@ bigtable:
 catalog:
 
   # -- Whether to enable the REST catalog service.
-  enabled: false
+  enabled: true
 
   # -- Iceberg catalog settings.
   iceberg:
 
     # -- The default warehouse name. Required. This is just a symbolic name; it must refer to a
     # declared warehouse below.
-    defaultWarehouse: ~  # warehouse1
+    defaultWarehouse: warehouse1
 
     # -- Iceberg config defaults applicable to all clients and warehouses. Any properties that are
     # common to all iceberg clients should be included here. They will be passed to all clients on
@@ -166,7 +166,7 @@ catalog:
     # warehouse must be defined.
     warehouses:
       # -- Symbolic name of the warehouse. Required.
-    - name: ~  # warehouse1
+    - name: warehouse1
       # -- Location of the warehouse. Required. Used to determine the base location of a table.
       # Scheme must be either s3 (Amazon S3), gs (Google GCS) or abfs / abfss (Azure ADLS). Storage
       # properties for each location can be defined below.
@@ -249,7 +249,7 @@ catalog:
         # -- Settings only relevant when clientAuthenticationMode is ASSUME_ROLE.
         serverIam:
           # -- Whether to enable server assume-role functionality.
-          enabled: ~  # false
+          enabled: true  # false
           # -- The ARN of the role to assume for accessing S3 data. This parameter is required for
           # Amazon S3, but may not be required for other storage providers (e.g. Minio does not use it
           # at all).
@@ -271,11 +271,11 @@ catalog:
         # -- AWS credentials. Required when serverAuthenticationMode is STATIC.
         accessKeySecret:
           # -- The secret name to pull AWS credentials from.
-          name: ~
+          name: k8sSecABC
           # -- The secret key storing the AWS secret key id.
-          awsAccessKeyId: ~
+          awsAccessKeyId: key111
           # -- The secret key storing the AWS secret access key.
-          awsSecretAccessKey: ~
+          awsSecretAccessKey: access111
 
       # -- Per-bucket S3 settings. Override the general settings above.
       buckets: []
```
CLI: `helm template --debug --namespace nessie-ns .`

Reply:
Hrm - that stuff won't become my friend.
Anyway, it should work now. Added a new `ci/catalog-secrets-values.yaml` and a new step to CI to run this `helm template` against each of those.

Reply:
> should reference nessie-catalog-secrets.s3.default-options.access-key from the config map as nessie.catalog.service.s3.default-options.access-key=nessie-catalog-secrets.s3.default-options.access-key

it is there:
```yaml
            #
            # s3.default-options.access-key
            #
            - name: "nessie.catalog.service.s3.default-options.access-key"
              value: "nessie-catalog-secrets.s3.default-options.access-key"
            - name: "nessie-catalog-secrets.s3.default-options.access-key.name"
              valueFrom:
                secretKeyRef:
                  name: "defaultCreds"
                  key: "defaultAccessKeyId"
            - name: "nessie-catalog-secrets.s3.default-options.access-key.secret"
              valueFrom:
                secretKeyRef:
                  name: "defaultCreds"
                  key: "defaultSecretAccessKey"#
```

via `helm template --debug --namespace nessie-ns --values ci/catalog-secrets-values.yaml .`

Reply:
Finally!!!
Got something into `ci.yml` that validates all this using "real" "secrets".